staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate_test.go

```diff
@@ -189,3 +189,132 @@ func TestFeatureGateFlagDefaults(t *testing.T) {
 		t.Errorf("Expected true")
 	}
 }
+
+func TestFeatureGateSetFromMap(t *testing.T) {
+	// gates for testing
+	const testAlphaGate Feature = "TestAlpha"
+	const testBetaGate Feature = "TestBeta"
+
+	tests := []struct {
+		setmap      map[string]bool
+		expect      map[Feature]bool
+		setmapError string
+	}{
+		{
+			setmap: map[string]bool{
+				"TestAlpha": true,
+				"TestBeta":  true,
+			},
+			expect: map[Feature]bool{
+				testAlphaGate: true,
+				testBetaGate:  true,
+			},
+		},
+		{
+			setmap: map[string]bool{
+				"TestBeta": true,
+			},
+			expect: map[Feature]bool{
+				testAlphaGate: false,
+				testBetaGate:  true,
+			},
+		},
+		{
+			setmap: map[string]bool{
+				"TestAlpha": false,
+			},
+			expect: map[Feature]bool{
+				testAlphaGate: false,
+				testBetaGate:  false,
+			},
+		},
+		{
+			setmap: map[string]bool{
+				"TestInvaild": true,
+			},
+			expect: map[Feature]bool{
+				testAlphaGate: false,
+				testBetaGate:  false,
+			},
+			setmapError: "unrecognized key:",
+		},
+	}
+	for i, test := range tests {
+		pairs := []string{}
+		for k, v := range test.setmap {
+			pairs = append(pairs, fmt.Sprintf("%s=%t", k, v))
```

Comment:
maps are not ordered. So we get potentially changing test names.

Reply:
I would give the tests a better, descriptive name instead.

Reply:
@sttts Done Thanks